### ISSUE TITLE
[Bug]: recurrent_gated_delta_rule 稳定入口、非连续 state 与尾块场景验证不完整

### ISSUE THREAD

**Issue by weinachuan** [flashserve/flash-linear-attention-npu#362]
## 环境信息

- 仓库版本：`main` 分支 `17e24fddc`
- 产品：A2、A5
- 安装方式：一体化 wheel
- 调用入口：`fla_npu.ops.ascendc.recurrent_gated_delta_rule` 与 fast kernel launch example

## 最小复现

```python
import math
import torch

from fla_npu.ops.ascendc import recurrent_gated_delta_rule

device = torch.device("npu")
T, HK, HV, K, V = 2, 1, 2, 80, 96
q = torch.randn(T, HK, K, dtype=torch.bfloat16, device=device)
k = torch.randn_like(q)
v = torch.randn(T, HV, V, dtype=torch.bfloat16, device=device)
beta = torch.rand(T, HV, dtype=torch.bfloat16, device=device)
storage = torch.empty(HV, T, V, K, dtype=torch.float32, device=device)
state = storage.permute(1, 0, 2, 3)
seq = torch.tensor([0, T], dtype=torch.int32, device=device)
indices = torch.arange(T, dtype=torch.int32, device=device)
gk = -torch.rand(T, HV, K, dtype=torch.float32, device=device)

out = recurrent_gated_delta_rule(
    q,
    k,
    v,
    state,
    beta=beta,
    scale=1.0 / math.sqrt(K),
    actual_seq_lengths=seq,
    ssm_state_indices=indices,
    gk=gk,
)
```

## 完整报错

当前调用链和验证资产存在以下问题：

1. `aclnnRecurrentGatedDeltaRuleGetWorkspaceSize` 缺少精确 ctypes `argtypes`，包含 `float` 属性的 ABI 参数无法稳定封送；未加载 `torch_npu` 时，多维 tensor descriptor 还可能回退为非 ND format，导致稳定入口启动失败。
2. FP32 `state` 的输出 dtype 推导不正确，tiling processor 保存临时 context 引用，存在生命周期风险。
3. fast kernel launch 的 functional 路径会把非连续 `state` 转成连续布局，kernel 也没有取得原始 stride，导致输出布局或状态寻址不符合输入语义。
4. 非整块 K/V 尾维和较短向量的计算依赖整向量接口，内存检查会暴露尾块越界读取风险；A2 的 add-fold reduce 对非整块 K 也不能正确覆盖实际有效长度。
5. 现有 ATK 目录只有单条占位用例，未覆盖 FP32 state、`gk`、双 gate、GQA、变长、无效前缀、accepted token、非连续 state、尾维和上限边界，也没有同时比较原地更新后的 state。

## 期望行为

稳定入口应按现有 aclnn ABI 正确调用算子，所有输入输出使用 ND descriptor，FP32 state 保持 FP32 并正确原地更新；fast kernel launch 应保留受支持的非连续 state stride。ATK 应通过 CPU 高精度/同精度双标杆同时验证 `out` 与 `state`，并保证 README 中 accuracy、performance、determinism、mssanitizer 四种调用在 A2、A5 上通过。

## 其他补充

公开 PR #314 的稳定入口修复方向和 PR #317 的非连续 state 修复方向均正确，但两者尚未合并，且缺少统一的 ATK 覆盖。建议在最新 `main` 上提交一个整合 PR，并由该 PR 统一替代 #314 和 #317。
